src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -1228,5 +1234,19 @@ private static bool EqualsOrdinal(string left, Span<byte> right)
                 _currentRequest?.GetHashCode() ?? 0,  // request ID
                 memberName,                   // method name
                 ToString() + ": " + message); // message
+
+        private sealed class MaxBytesAllowed
```

Comment:
Can we make this a struct and pass by ref as necessary?  That seems more intuitive.  Or is the problem that the methods are async?

Reply:
> is the problem that the methods are async?

Exactly